[Bug] Cannot enable auto-shift and HOLD_ON_OTHER_KEY_PRESS_PER_KEY at the same time

### Describe the Bug

The function prototype for `get_hold_on_other_key_press` is missing from `/quantum/action_tapping.h`. This usually doesn't cause a problem, even with `HOLD_ON_OTHER_KEY_PRESS_PER_KEY` enabled in `config.h`; however, with auto-shift also enabled, the following error is generated while compiling:

```
Compiling: quantum/process_keycode/process_auto_shift.c                                            quantum/process_keycode/process_auto_shift.c: In function 'process_auto_shift':
quantum/process_keycode/process_auto_shift.c:406:23: error: implicit declaration of function 'get_hold_on_other_key_press' [-Werror=implicit-function-declaration]
  406 |                     ? get_hold_on_other_key_press(keycode, record)
      |                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1.exe: all warnings being treated as errors
```

Adding the prototype in `action_tapping.h` with the other mod-tap prototypes fixes the error.

### Keyboard Used

_No response_

### Link to product page (if applicable)

_No response_

### Operating System

_No response_

### qmk doctor Output

_No response_

### Is AutoHotKey / Karabiner installed

- [ ] AutoHotKey (Windows)
- [ ] Karabiner (macOS)

### Other keyboard-related software installed

_No response_

### Additional Context

_No response_

--- a/quantum/action_tapping.h
+++ b/quantum/action_tapping.h
@@ -40,6 +40,7 @@ bool     get_permissive_hold(uint16_t keycode, keyrecord_t *record);
 bool     get_ignore_mod_tap_interrupt(uint16_t keycode, keyrecord_t *record);
 bool     get_tapping_force_hold(uint16_t keycode, keyrecord_t *record);
 bool     get_retro_tapping(uint16_t keycode, keyrecord_t *record);
+bool     get_hold_on_other_key_press(uint16_t keycode, keyrecord_t *record);
 
 #ifdef DYNAMIC_TAPPING_TERM_ENABLE
 extern uint16_t g_tapping_term;

